tests/test_controller.py

```diff
@@ -445,7 +445,7 @@ def mock_for_library(cls, incoming_library):
         # We also set up some patron web client settings that will
         # be loaded.
         ConfigurationSetting.sitewide(
-            self._db, Configuration.PATRON_WEB_CLIENT_URL).value = "http://sitewide/1234"
+            self._db, Configuration.PATRON_WEB_HOSTNAMES).value = list("http://sitewide/1234")
```

Comment:
This is being written to the database, so you can't set it to a list value; use json.dumps() to convert the list into a string first.

A couple lines below this, we set a per-library value for this setting (as we discussed, something that's not currently possible through the UI), and that value is set to a string instead of a list. For reasons I don't understand, this works fine -- probably the code that parses this value into a JSON list falls back to "a list containing a single string" if it can't parse the value. But that bit should also be changed.

I think the failure in `tests.test_custom_index.TestCustomIndexView.test_for_library` is caused by the failure of `test_load_settings`, which mocks some system code and crashes before it can un-mock it. When you fix this test the other failure should go away.

You can run just the failing tests like so:

```
nosetests tests/test_controller.py:TestCirculationManager.test_load_settings tests/test_custom_index.py:TestCustomIndexView.test_for_library
```

Reply:
Thank you for the details here. Was just digging in to this, v helpful